worlds/tits_the_3rd/client/client.py

```diff
@@ -125,14 +130,16 @@ def on_package(self, cmd: str, args: dict):
             if not self.install_game_mod():
                 raise Exception("Error Installing Game Mod")
             logger.info("Finished Installing Game Mod")
-            self.game_interface = TitsThe3rdMemoryIO(self.exit_event)
 
             asyncio.create_task(self.send_msgs([{"cmd": "LocationScouts", "locations": self.location_ids}]))
 
         elif cmd == "LocationInfo":
-            for item in [NetworkItem(*item) for item in args["locations"]]:
-                if self.player_names[item.player] != self.slot_info[self.slot].name:
-                    self.non_local_locations.add(item.location)
+            if not self.non_local_locations_initiated:
+                for item in [NetworkItem(*item) for item in args["locations"]]:
+                    if self.player_names[item.player] != self.slot_info[self.slot].name:
+                        self.non_local_locations.add(item.location)
+                self.non_local_locations_initiated = True
+                self.game_interface = TitsThe3rdMemoryIO(self.exit_event)
```

Comment:
Does the memory interface need to be initiated after we have non_location_locations set?

Reply:
I'm thinking this is kind of a confusing place to stick this initialization. Its unclear without context when LocationInfo is called or why we need to init the game interface here. What are your thoughts to adding a `init_game_interface` method which is called after these lines in the client loop:

```
        if not ctx.server:
            # client disconnected from server
            ctx.reset_client_state()
            await ctx.wait_for_ap_connection()
            continue
        
        if not ctx.game_interface:
            init_game_interface
```

This also asserts that we received all the data in ctx.wait_for_ap_connection.

We should also stick any wait conditions behind `client_received_initial_server_data` (e.g.` self.non_local_locations_initiated`)

Reply:
yeah make sense